test/caesium/crypto/pwhash_test.clj

```diff
@@ -0,0 +1,149 @@
+(ns caesium.crypto.pwhash-test
+  (:require [caesium.crypto.pwhash :as p]
+            [caesium.byte-bufs :as bb]
+            [caesium.test-utils :refer [const-test]]
+            [caesium.vectors :as v]
+            [caesium.util :as u]
+            [caesium.randombytes :as r]
+            [clojure.test :refer [are deftest is]]))
+
+
+(const-test
+ p/alg-argon2i13 1
+ p/alg-default 1
+ p/bytes-min 16
+ p/bytes-max 4294967295
+ p/passwd-min 0
+ p/passwd-max 4294967295
+ p/saltbytes 16
+ p/strbytes 128
+ p/strprefix "$argon2i$"
+ p/opslimit-min 3
+ p/opslimit-max 4294967295
+ p/memlimit-min 8192
+ p/opslimit-interactive 4
+ p/memlimit-interactive  33554432
+ p/opslimit-moderate 6
+ p/memlimit-moderate 134217728
+ p/opslimit-sensitive 8
+ p/memlimit-sensitive 536870912
+
+ p/argon2i-alg-argon2i13 1
+ p/argon2i-bytes-min 16
+ p/argon2i-bytes-max 4294967295
+ p/argon2i-passwd-min 0
+ p/argon2i-passwd-max 4294967295
+ p/argon2i-saltbytes 16
+ p/argon2i-strbytes 128
+ p/argon2i-strprefix "$argon2i$"
+ p/argon2i-opslimit-min 3
+ p/argon2i-opslimit-max 4294967295
+ p/argon2i-memlimit-min 8192
+ p/argon2i-opslimit-interactive 4
+ p/argon2i-memlimit-interactive  33554432
+ p/argon2i-opslimit-moderate 6
+ p/argon2i-memlimit-moderate 134217728
+ p/argon2i-opslimit-sensitive 8
+ p/argon2i-memlimit-sensitive 536870912
+
+ p/argon2id-alg-argon2id13 2
+ p/argon2id-bytes-min 16
+ p/argon2id-bytes-max 4294967295
+ p/argon2id-passwd-min 0
+ p/argon2id-passwd-max 4294967295
+ p/argon2id-saltbytes 16
+ p/argon2id-strbytes 128
+ p/argon2id-strprefix "$argon2id$"
+ p/argon2id-opslimit-min 1
+ p/argon2id-opslimit-max 4294967295
+ p/argon2id-memlimit-min 8192
+ p/argon2id-opslimit-interactive 2
+ p/argon2id-memlimit-interactive  67108864
+ p/argon2id-opslimit-moderate 3
+ p/argon2id-memlimit-moderate 268435456
+ p/argon2id-opslimit-sensitive 4
+ p/argon2id-memlimit-sensitive 1073741824
+
+ p/scryptsalsa208sha256-bytes-min 16
+ p/scryptsalsa208sha256-bytes-max 0x1fffffffe0
+ p/scryptsalsa208sha256-passwd-min 0
+ p/scryptsalsa208sha256-saltbytes 32
+ p/scryptsalsa208sha256-strbytes 102
+ p/scryptsalsa208sha256-strprefix "$7$"
+ p/scryptsalsa208sha256-opslimit-min 32768
+ p/scryptsalsa208sha256-opslimit-max 4294967295
+ p/scryptsalsa208sha256-memlimit-min 16777216
+ p/scryptsalsa208sha256-opslimit-interactive 524288
+ p/scryptsalsa208sha256-memlimit-interactive  16777216
+ p/scryptsalsa208sha256-opslimit-sensitive 33554432
+ p/scryptsalsa208sha256-memlimit-sensitive  1073741824)
+
+(deftest pwhash-alg-default-test
+  []
+  (let [salt (u/unhexify "7cb3b8ceb58e7847fc4485e63dbfdb9b")]
+    (is (= "75a10fdb4db0836498f824f1f0cc9ab9d3bb194d41b8dd66bd1ca6f0cf686810" (u/hexify (p/pwhash 32 "password" salt p/opslimit-min p/memlimit-interactive p/alg-default))))))
```

Comment:
Where did these consts come from?

(I think I've asked this before, but it's important that it's on the PR :))

Reply:
I looked up the constants in the include file in libsodium source. However, there are some that are not straight forward (sodium calculates them on the fly). I did this mainly to make sure that my code was correctly "linked" to libsodium.

Reply:
You should probably put a comment there to that effect.